Makefile

```diff
@@ -138,34 +135,14 @@ generate-clientset: .remove-clientset .remove-deepcopy .remove-crds
 .remove-clientset:
 	rm -rf $(CURDIR)/$(PKG_BASE)/clientset
 
-# Generate an image for containerized builds
-# Note: This image is local only
-.PHONY: .build-image .pull-build-image .push-build-image
-.build-image: docker/Dockerfile.devel
-	if [ x"$(SKIP_IMAGE_BUILD)" = x"" ]; then \
-		$(DOCKER) build \
-			--progress=plain \
-			--build-arg GOLANG_VERSION="$(GOLANG_VERSION)" \
-			--tag $(BUILDIMAGE) \
-			-f $(^) \
-			docker; \
-	fi
-
-.pull-build-image:
-	$(DOCKER) pull $(BUILDIMAGE)
-
-.push-build-image:
-	$(DOCKER) push $(BUILDIMAGE)
-
-$(DOCKER_TARGETS): docker-%: .build-image
-	@echo "Running 'make $(*)' in docker container $(BUILDIMAGE)"
+$(DOCKER_TARGETS): docker-%: 
+	@echo "Running 'make $(*)' in container image $(BUILDIMAGE)"
 	$(DOCKER) run \
 		--rm \
-		-e HOME=$(PWD) \
-		-e GOCACHE=$(PWD)/.cache/go \
-		-e GOPATH=$(PWD)/.cache/gopath \
-		-v $(PWD):$(PWD) \
-		-w $(PWD) \
+		-e GOCACHE=/tmp/.cache/go \
+		-e GOMODCACHE=/tmp/.cache/gopath \
```

Comment:
OK. Now I see the confusion. We no longer set `GOPATH` here. Which of these need to be set?

Reply:
GOPATH is `/go` by default in the `golang` images
```bash
root@83d456ef8ffe:/go# env
HOSTNAME=83d456ef8ffe
PWD=/go
HOME=/root
GOLANG_VERSION=1.21.6
TERM=xterm
SHLVL=1
GOTOOLCHAIN=local
PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
GOPATH=/go
_=/usr/bin/env
```

Reply:
so here we are only defining the ones we want to modify